users/vieting/experiments/switchboard/ctc/feat/baseline_args.py

```diff
@@ -237,6 +237,54 @@ def get_returnn_config(
         # network["source"] = specaug_layer_jingjing(in_layer=["features"])
         pass
 
+    if audio_perturbation and recognition:
+        feature_net = copy.deepcopy(feature_net)
+        audio_perturb_args = extra_args.get("audio_perturb_args", {})
+        source_layer = "data"
+        if "preemphasis" in audio_perturb_args:
+            for layer in feature_net["subnetwork"]:
+                if feature_net["subnetwork"][layer].get("from", None) == source_layer:
+                    feature_net["subnetwork"][layer]["from"] = "preemphasis"
+                    source_layer = "preemphasis"
+            alpha = audio_perturb_args.get("default")
+            feature_net["subnetwork"]["preemphasis"] = PreemphasisNetwork(alpha=alpha).get_as_subnetwork()
+        if "codecs" in audio_perturb_args:
+            for layer in feature_net["subnetwork"]:
+                if feature_net["subnetwork"][layer].get("from", None) == source_layer:
+                    feature_net["subnetwork"][layer]["from"] = "codec"
+                    source_layer = "codec"
```

Comment:
It has to be done after looping through the layers, right?
```suggestion
            source_layer = "codec"
```